invalid memory address or nil pointer dereference

### What happened?

script [spec-to-yaml.sh](https://github.com/kubernetes/kubernetes/blob/master/test/conformance/spec-to-yaml.sh) result to a nil pointer dereference exception

e.g.
`go run ./test/conformance/walk.go --source=/go/src/k8s.io/kubernetes ./_output/specsummaries.json > ./_output/conformance.yaml`

show the below exception, 
```
Did not find test info for spec: "[sig-apps] Deployment deployment should support rollover [Conformance]"
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x20 pc=0x5622c0]
```



### What did you expect to happen?

no exception found when the script got called.

### How can we reproduce it (as minimally and precisely as possible)?

1. Generate a `e2e.test` test binary (which are a little behind the master)
2. Dump the spec on the latest master branch, https://github.com/kubernetes/kubernetes/blob/master/test/conformance/gen-specsummaries.sh
3. Convert the spec to yaml: https://github.com/kubernetes/kubernetes/blob/master/test/conformance/spec-to-yaml.sh

 

### Anything else we need to know?

_No response_

### Kubernetes version

<details>

```console
$ kubectl version
master branch
# paste output here
```

</details>


### Cloud provider

<details>

</details>
N/A

### OS version

<details>

```console
# On Linux:
$ cat /etc/os-release
# paste output here
$ uname -a
# paste output here

# On Windows:
C:\> wmic os get Caption, Version, BuildNumber, OSArchitecture
# paste output here
```

</details>


### Install tools

<details>

</details>


### Container runtime (CRI) and version (if applicable)

<details>

</details>


### Related plugins (CNI, CSI, ...) and versions (if applicable)

<details>

</details>

--- a/test/conformance/walk.go
+++ b/test/conformance/walk.go
@@ -108,10 +108,9 @@ func main() {
 			testInfo := getTestInfo(spec)
 			if testInfo != nil {
 				testInfos = append(testInfos, testInfo)
-			}
-
-			if err := validateTestName(testInfo.CodeName); err != nil {
-				log.Fatal(err)
+				if err := validateTestName(testInfo.CodeName); err != nil {
+					log.Fatal(err)
+				}
 			}
 		}
 	}

